packaging/greenboot/microshift-running-check.sh

```diff
@@ -128,15 +128,22 @@ for i in "${!PODS_NS_LIST[@]}"; do
 done
 
 # Verify that MicroShift core pods are not restarting
-pids=()
+declare -A pid2name
 for i in "${!PODS_NS_LIST[@]}"; do
     CHECK_PODS_NS=${PODS_NS_LIST[${i}]}
 
     echo "Checking pod restart count in the '${CHECK_PODS_NS}' namespace"
     namespace_pods_not_restarting "${CHECK_PODS_NS}" &
-    pids+=($!)
+    pid=$!
+
+    pid2name["${pid}"]="${CHECK_PODS_NS}"
 done
 
-for pid in "${pids[@]}"; do
-    wait "${pid}"
+# Wait for the restart check functions to complete, printing errors in case of a failure
+for pid in "${!pid2name[@]}"; do
+    if ! wait "${pid}" ; then
+        name=${pid2name["${pid}"]}
+        echo "Pod restart count check failed for the '${name}' namespace"
+        exit 1
```

Comment:
if there's a problem with many namespaces, this will cause to print out first one.

do you think it might be better to print all problematic namespaces so it's easier to see if the problem is "global" or it's just one namespace?